src/cli.rs

```diff
@@ -37,6 +43,27 @@ pub struct Opt {
     pub mine_when_not_synced: bool,
 }
 
+fn parse_devfund_percent(s: &str) -> Result<u16, &'static str> {
+    let err = "devfund-percent should be --devfund-percent=XX.YY up to 2 numbers after the dot";
+    let mut splited= s.split('.');
+    let prefix = splited.next().ok_or(err)?;
+    // if there's no postfix then it's 0.
+    let postfix = splited.next().ok_or(err).unwrap_or("0");
+    // error if there's more than a single dot
+    if splited.next().is_some() {return Err(err)};
+    // error if there are more than 3 numbers before the dot(100) or 2 numbers after
+    if prefix.len() > 3 || postfix.len() > 2 {
+        return Err(err);
+    }
+    let postfix: u16 = postfix.parse().map_err(|_|err)?;
+    let prefix: u16 = prefix.parse().map_err(|_|err)?;
+    // can't be more than 100%,
+    if prefix > 100 || postfix > 100 || (prefix == 100 && postfix != 0) {
```

Comment:
Can simply be `prefix >= 100 || postfix > 100` because you don't need to support 100% devfund, one can do that by passing a single mining address to start with

Reply:
hmm, yeah I wanted to support 100%, but you're right it doesn't make sense